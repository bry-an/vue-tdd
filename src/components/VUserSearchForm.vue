<template>
    <form @submit.prevent="$emit('submitted', username)">
        <input type="text" v-model="username" />
        <button type="submit">Send</button>
    </form>
</template>

<script>
export default {
  name: 'VUserSearchForm', 
  data() {
      return {
          username: ''
      }
  }
}
</script>