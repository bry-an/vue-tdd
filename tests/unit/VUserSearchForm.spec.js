import { shallowMount } from "@vue/test-utils"
import VUserSearchForm from "@/components/VUserSearchForm"

describe("VUserSearchForm", () => {
    const build = () => {
        const wrapper = shallowMount(VUserSearchForm)

        return {
            wrapper, 
            input: () => wrapper.find("input"),
            button: () => wrapper.find("button")
        }
    }
    
    it("renders the component", () => {
        // arrange
       const { wrapper } = build()
       
       expect(wrapper.html()).toMatchSnapshot()

    })

    it("renders main child components", () => {
        // arrange
        const { input, button } = build()

        expect(input().exists()).toBe(true)
        expect(button().exists()).toBe(true)
    })

    it("calls 'submitted' event when submitting form", () => {
        const expectedUser = 'bryan'
        const { wrapper, button, input } = build()
        input().element.value = expectedUser // manually entering input value

        // do 
        input().trigger("input")
        button().trigger("click")
        button().trigger("submit")

        // assert
        expect(wrapper.emitted().submitted[0]).toEqual([expectedUser])
    })

})