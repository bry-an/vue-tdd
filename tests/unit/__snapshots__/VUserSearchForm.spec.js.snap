// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserSearchForm renders the component 1`] = `
<div>
  UserSearchForm
</div>
`;
